server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -121,7 +122,8 @@ public IndicesClusterStateService(
         final SnapshotShardsService snapshotShardsService,
         final PrimaryReplicaSyncer primaryReplicaSyncer,
         final RetentionLeaseSyncer retentionLeaseSyncer,
-        final NodeClient client
+        final NodeClient client,
+        final IndicesWriteLoadStatsCollector indicesWriteLoadStatsCollector
```

Comment:
I wonder if we should pass in the service instead here and let it delegate to the collector? Seems like we have the service orchestrating most things except this bit. That could also avoid registering the collector in Node.